lib/faucet_pipeline_rails/manifest.rb

```diff
@@ -5,32 +5,33 @@ class Manifest
     include Singleton
 
     def fetch(asset_name)
-      parsed_manifest.fetch(asset_name)
+      manifest.fetch(asset_name)
     rescue KeyError
       raise "The asset '#{asset_name}' was not in the manifest"
     end
 
     private
 
+    def manifest
+      return parsed_manifest unless Rails.env.production?
+      @manifest ||= parsed_manifest
+    end
+
     def parsed_manifest
       JSON.parse(unparsed_manifest)
     rescue JSON::ParserError
-      raise "The manifest file '#{relative_manifest_path}' is invalid JSON"
+      raise "The manifest file '#{manifest_path}' is invalid JSON"
     end
 
     def unparsed_manifest
       File.read(manifest_path)
     rescue Errno::ENOENT
-      raise "The manifest file '#{relative_manifest_path}' is missing"
+      raise "The manifest file '#{manifest_path}' is missing"
     end
 
     def manifest_path
-      File.join(Rails.root, relative_manifest_path)
-    end
-
-    def relative_manifest_path
       Rails.configuration.x.faucet_pipeline.manifest_path ||
-        File.join("public", "assets", "manifest.json")
+        Rails.root.join("public", "assets", "manifest.json")
```

Comment:
IIRC the idiomatic way of configuring this would be to add the config option and its default into the `FaucetPipelineRails::Railtie`:

```ruby
class FaucetPipelineRails::Railtie < Rails::Railtie
  config.x.faucet_pipeline.manifest_path = Rails.root.join("public", "assets", "manifest.json")
end
```